### DISCUSSION TITLE
Literate Programming

### DISCUSSION THREAD

**Discussion by AaronDovTurkel** [modernweb-dev/rocket#378] [Ideas]
HI 👋 ,
I would love to see a way to harness Rocket and maybe unifiedJS to produce source code as well as documentation similar to [Donald Knuth's Literate Programming tool WEB](https://en.wikipedia.org/wiki/Literate_programming). This could even be integrated into the web.dev platform in general.

**Comment by daKmoR** [comment]
hey 👋,

that sounds like a good idea 👍

what would we need for that? anything you can pick up - or we can pick up together? feel free to share some ideas - I totally open for crazy plans 😬

**Reply by AaronDovTurkel** [reply]
Hey @daKmoR,
I would imagine starting off with something simple like [ganesha](https://github.com/hackbg/ganesha) where each file is just a markdown file with a LS bridge for debugging. This way you can compile the entire source code into either documentation or AST.

**Reply by daKmoR** [reply]
hmm I'm not sure I understand what the end goal is 🤔

How will it be different to [vs code notebooks](https://code.visualstudio.com/blogs/2021/11/08/custom-notebooks)?

Care to elaborate in more detail on which user problem it would solve? and what a typical user flow would look like.

PS: VS Code just released it [markdown language server](https://code.visualstudio.com/blogs/2022/08/16/markdown-language-server) maybe something worth considering

**Reply by AaronDovTurkel** [reply]
I was thinking of a development workflow that would involve just writing markdown files with code snippets...
So for instance, if I were developing a React application, my components folder would consist of just markdown files with the javascript code in code blocks. The goal of integrating with UnifiedJS and/or Rocket would be to be able to compile that code already from the markdown file. Eventually, I can see the whole application just being a set of markdown files with a readily producible doc book to boot. I wonder if the Markdown server would help in this instance because we sort of need to parse the markdown for code blocks and compile the code blocks at the same time (if running in a watch mode for instance). Something like [Eve](http://witheve.com/deepdives/literate.html) .

**Comment by daKmoR** [comment]
Thanks for opening this discussion and for helping improve Rocket.

Rocket has been rebuilt from a clean slate for the new @rocket/js release. The implementation, docs, APIs, and project structure have changed substantially, so the old issue history no longer maps cleanly to the current codebase.

I’m closing this as part of the repository reset.

This is currently a soft launch: the package and docs are live, and we’re dogfooding the new version on real projects before doing a larger public relaunch.

If this discussion is still relevant with the new Rocket, please open a fresh discussion/issue against the current main branch with updated context.

Thanks again.
